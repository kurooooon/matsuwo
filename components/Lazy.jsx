import React from 'react';
import { InView } from 'react-intersection-observer'

const Lazy = function Lazy({ children, rootMargin, triggerOnce }) { 
  return (
    <InView triggerOnce={triggerOnce} rootMargin={rootMargin}>
      {({ inView, ref }) => (
        <div ref={ref}>{inView ? children : null}</div>
      )}
    </InView>
  );
}

export default Lazy