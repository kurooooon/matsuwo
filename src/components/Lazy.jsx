import React from 'react';
import { useInView } from 'react-intersection-observer'

const Lazy = ({ children, rootMargin, triggerOnce }) => { 
  const [ref, inView] = useInView({
    rootMargin,
    triggerOnce
  });
  return (
    <div ref={ref}>{inView ? children : null}</div>
  );
}

export default Lazy