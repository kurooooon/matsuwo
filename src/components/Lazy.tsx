import React from 'react';
import { useInView } from 'react-intersection-observer';

type Props = {
  children: React.ReactNode;
  rootMargin: string;
  triggerOnce: boolean;
};

const Lazy = ({ children, rootMargin, triggerOnce }: Props) => {
  const [ref, inView] = useInView({
    rootMargin,
    triggerOnce,
  });
  return <div ref={ref}>{inView ? children : null}</div>;
};

export default Lazy;
