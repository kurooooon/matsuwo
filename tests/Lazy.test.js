import React from 'react';
import Lazy from '../components/Lazy';
import renderer from 'react-test-renderer';
import { mockAllIsIntersecting } from 'react-intersection-observer/test-utils'

test('<Lazy> snapshot testing', () => {
  const component = renderer.create(
    <Lazy>test</Lazy>
  );
  mockAllIsIntersecting(true);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});