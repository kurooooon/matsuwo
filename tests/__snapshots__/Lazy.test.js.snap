// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Lazy> snapshot testing 1`] = `<div />`;
